src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -65,7 +69,12 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
         dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
         dispatch({ type: WalletActions.SET_IS_LOCKED, payload: isLocked })
         setLocalWalletTypeAndDeviceId(KeyManager.MetaMask, deviceId)
-        dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+        if (snapIsInstalled) {
+          dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+        } else {
+          history.push('/metamask/snap/install')
+        }
+        // if it is installed ask them if they would like to use it
```

Comment:
I reckon this should automagically happen? I can't think of a case where you have the snap installed, enabled, but don't expect it to automagically work - since a snap is effectively installed for a specific domain and will automagically work there